src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+
+    @Override
+    public void add(Order order) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(order);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add order " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery("SELECT o FROM Order o"
+                    + " LEFT JOIN FETCH o.user  "
+                    + " LEFT JOIN FETCH o.tickets t "
+                    + " LEFT JOIN FETCH t.movieSession ms "
+                    + " LEFT JOIN FETCH ms.movie m "
+                    + " LEFT JOIN FETCH ms.cinemaHall cmh "
+                    + " WHERE o.user = :user", Order.class);
+            query.setParameter("user", user);
+            return query.getResultList();
+        } catch (RuntimeException e) {
+            throw new DataProcessingException("Can't get order by user " + user, e);
```

Comment:
Consider making the exception message more descriptive. For example, "Can't retrieve orders for user " + user.getId() + " from the database." This provides more context about the operation that failed.